### PR TITLE
Add METL(math expression template library)

### PR DESCRIPTION
**Pull request by TillHeinzel**
Hi. I added my parser METL to the benchmark. It is only a few months old and not super mature, but doing reasonably well in the benchmark.
It compiles under vs 2017, but I haven't tested it under 2015. You may want to ignore the vs-solution, which I had to upgrade. I also didn't make any changes to the Qt-creator project.

### CODE REVIEW

**Comment by ArashPartow** [comment]
@TillHeinzel Thanks for the PR, got a few comments:

1. Can you please remove the **gitignore** file

2. Can you please provide a run of **_"bench_expr_all.txt"_** with your parser included as a comment in this thread

3. What do you expect the result of the following expression will be: _e^e^2_ ? 

4. Does your parser pass all the tests in **_"bench_expr_weird.txt"_** ?  _(optional not a show stopper)_

5. Does your parser or its dependencies require C++11 or later?   _(optional not a show stopper)_

File: src/BenchMETL.cpp [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		auto tests_list = test_expressions();
+
+		for (auto test : tests_list)
+		{
+			try
+			{
+				auto f = compiler.build<double>(test.first);
+				test_result = (f() == test.second);
+			}
+			catch (const std::runtime_error& err)
```
Comment (ArashPartow) [inline src/BenchMETL.cpp]: The **err** variable is unused, causes a warning to be emitted.

**Comment by TillHeinzel** [comment]
Hi again, sorry for the wait. 
All right:
1. gitignore removed

2. shootout for all
[Shootout_20180408_075537.txt](https://github.com/ArashPartow/math-parser-benchmark-project/files/1887077/Shootout_20180408_075537.txt)

3. shootout for weird (METL passes all tests)
[Shootout_20180408_080038.txt](https://github.com/ArashPartow/math-parser-benchmark-project/files/1887078/Shootout_20180408_080038.txt)

4. METL requires C++14. You could probably get by with just C++11 by removing addDefaults (uses template lambdas), but I haven't tried and there are probably some C++14 features in there somewhere that I don't consider anything special anymore.

5. removed err-variable

**Comment by ArashPartow** [comment]
@TillHeinzel Thanks for the update, just one quick question, wrt Shootout_20180408_075537.txt, it seems the timings for ExprTk never go below 10ns - I believe there is something wrong/problematic in your run. 

As an example, given the first expression "1",  it is simply a literal value return, it should execute in a time similar to that of the ExprTkFloat variant.

**Comment by TillHeinzel** [comment]
I agree, that makes no sense. Especially because I have been wondering how you magiced your parser to be so fast! 

I have done another run, and this time it seems like nothing went wrong:
all:
[Shootout_20180419_074300.txt](https://github.com/ArashPartow/math-parser-benchmark-project/files/1926776/Shootout_20180419_074300.txt)

weird:
[Shootout_20180419_074235.txt](https://github.com/ArashPartow/math-parser-benchmark-project/files/1926775/Shootout_20180419_074235.txt)

But you will probably do the run yourself again anyway, right?

**Comment by ArashPartow** [comment]
@TillHeinzel  That looks much better!

btw wrt to the following comment in **BenchMETL.cpp**: 

> // not sure I get what this is for

This is done so as to reset the various CPU counters used for OOO, _predictive branching/execution_ that may have annealed from the previous parser running the benchmark. 

**In short:** make sure the previous parser doesn't _interfere_ with the current parser's run.

**Comment by TillHeinzel** [comment]
Interesting, and sensible. Not something I would have thought of ;) I will just remove my comment. Is there anything else standing in the way of merging?

**Comment by ArashPartow** [comment]
@TillHeinzel   I've merged METL into the benchmark, made some minor changes. 

Thank-you for contributing your parser to the benchmark suite.

**Comment by TillHeinzel** [comment]
@ArashPartow Thank you for adding it :)

**Comment by TillHeinzel** [comment]
@ArashPartow  Wow, that looks a lot better than in my runs! Probably because of the number of iterations. I was really going for flexibility, not speed, but it seems like there is some of that in there, as well! Not as fast as ExprTK, of course, but I am exceedingly happy with this result!

If I make any major changes to METL, I will be sure to update the benchmark and send a pull request :)

Thanks for making this benchmark-project, it was really easy to add METL and see how it compares to other libraries. It even helped me find a few bugs.

**Pull request closed without merging**